hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ClusterStateEndpoint.java

```diff
@@ -0,0 +1,128 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Endpoint to fetch current state of ozone cluster.
+ */
+@Path("/clusterState")
+@Produces(MediaType.APPLICATION_JSON)
+public class ClusterStateEndpoint {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(ClusterStateEndpoint.class);
+
+  private ReconNodeManager nodeManager;
+  private ReconPipelineManager pipelineManager;
+  private ReconContainerManager containerManager;
+  private ReconOMMetadataManager omMetadataManager;
+
+  @Inject
+  ClusterStateEndpoint(OzoneStorageContainerManager reconSCM,
+                       ReconOMMetadataManager omMetadataManager) {
+    this.nodeManager =
+        (ReconNodeManager) reconSCM.getScmNodeManager();
+    this.pipelineManager = (ReconPipelineManager) reconSCM.getPipelineManager();
+    this.containerManager =
+        (ReconContainerManager) reconSCM.getContainerManager();
+    this.omMetadataManager = omMetadataManager;
+  }
+
+  /**
+   * Return a summary report on current cluster state.
+   * @return {@link Response}
+   */
+  @GET
+  public Response getClusterState() {
+    List<DatanodeDetails> datanodeDetails = nodeManager.getAllNodes();
+    AtomicInteger healthyDatanodes = new AtomicInteger();
+    int containers = this.containerManager.getContainerIDs().size();
+    int pipelines = this.pipelineManager.getPipelines().size();
+    long volumes;
+    long buckets;
+    long keys;
+    AtomicLong capacity = new AtomicLong(0L);
+    AtomicLong used = new AtomicLong(0L);
+    AtomicLong remaining = new AtomicLong(0L);
+    datanodeDetails.forEach(datanode -> {
+      NodeState nodeState = nodeManager.getNodeState(datanode);
+      SCMNodeStat nodeStat = nodeManager.getNodeStat(datanode).get();
+      if (nodeState.equals(NodeState.HEALTHY)) {
+        healthyDatanodes.getAndIncrement();
+      }
+      capacity.getAndAdd(nodeStat.getCapacity().get());
+      used.getAndAdd(nodeStat.getScmUsed().get());
+      remaining.getAndAdd(nodeStat.getRemaining().get());
+    });
+    DatanodeStorageReport storageReport =
+        new DatanodeStorageReport(capacity.get(), used.get(), remaining.get());
+    DatanodesCount datanodesCount = new DatanodesCount(datanodeDetails.size(),
+        healthyDatanodes.get());
+    ClusterStateResponse.Builder builder = ClusterStateResponse.newBuilder();
+    try {
+      volumes = omMetadataManager.getVolumeTable().getEstimatedKeyCount();
+      builder.setVolumes(volumes);
+    } catch (Exception ex) {
+      LOG.error("Unable to get Volumes count in ClusterStateResponse.", ex);
+    }
+    try {
+      buckets = omMetadataManager.getBucketTable().getEstimatedKeyCount();
```

Comment:
Nit. We can directly use builder.set(omm.getTable(). getEstimatedKeyCount()). We don't need local variables. (Same for keys, volumes, pipelines, containers etc)